frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditLayoutSetForSubform/EditLayoutSet/CreateNewLayoutSet/CreateNewLayoutSet.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { StudioButton, StudioCard, StudioTextfield } from '@studio/components';
+import { ClipboardIcon, CheckmarkIcon } from '@studio/icons';
+import { useAddLayoutSetMutation } from 'app-development/hooks/mutations/useAddLayoutSetMutation';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import classes from './CreateNewLayoutSet.module.css';
+
+type CreateNewLayoutSetProps = {
+  onSubFormCreated: (layoutSetName: string) => void;
+};
+
+export const CreateNewLayoutSet = ({
+  onSubFormCreated,
+}: CreateNewLayoutSetProps): React.ReactElement => {
+  const { t } = useTranslation();
+  const [newSubForm, setNewSubForm] = useState('');
+  const { org, app } = useStudioEnvironmentParams();
+  const { mutate: addLayoutSet } = useAddLayoutSetMutation(org, app);
+
+  const createNewSubform = () => {
+    if (!newSubForm) return;
+    addLayoutSet({
+      layoutSetIdToUpdate: newSubForm,
+      layoutSetConfig: {
+        id: newSubForm,
+        type: 'subform',
+      },
+    });
+    onSubFormCreated(newSubForm);
+    setNewSubForm('');
+  };
+
+  function onNameChange(subFormName: string) {
+    setNewSubForm(subFormName);
+  }
+
+  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    onNameChange(e.target.value);
+  }
```

Comment:
Could we simplify this?

```suggestion
  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
    setNewSubForm(e.target.value);
  }
```

Reply:
fixed